packages/core/contracts/gnosis-zodiac/implementation/OptimisticGovernor.sol

```diff
@@ -215,39 +241,41 @@ contract OptimisticGovernor is Module, Lockable {
         // Create the proposal hash.
         bytes32 proposalHash = keccak256(abi.encode(_transactions));
 
-        // Add the proposal hash to ancillary data.
-        bytes memory ancillaryData = AncillaryData.appendKeyValueBytes32("", PROPOSAL_HASH_KEY, proposalHash);
+        // Add the proposal hash and explanation to ancillary data.
+        bytes memory claim = _constructClaim(proposalHash, _explanation);
 
-        // Check that the proposal is not already mapped to a proposal time, i.e., is not a duplicate.
-        require(proposalHashes[proposalHash] == 0, "Duplicate proposals are not allowed");
+        // Check that the proposal is not already mapped to an assertionId, i.e., is not a duplicate.
+        require(proposalHashes[proposalHash] == bytes32(0), "Duplicate proposals not allowed");
 
-        // Map the proposal hash to the current time.
-        proposalHashes[proposalHash] = time;
+        // Check the minimum required bond and use that instead if it is greater than the bondAmount.
+        uint256 minimumBond = optimisticAsserter.getMinimumBond(address(collateral));
+        uint256 totalBond = minimumBond > bondAmount ? minimumBond : bondAmount;
```

Comment:
the bond logic here seems a bit strange. The point of having the `bondAmount` set within the OG it to let the owner set a bond amount that is, in theory, much larger than the OA minimum, right?

Reply:
yes, owner of OG might require higher amount than required by OA. But if it is below we might not want this to block proposals.

Reply:
alternative would be to check that `bondAmount` meets the minimum bond when configuring OG, but this can become obsolete if final fee gets increased.